### ISSUE TITLE
GTM4WP - Category product links take seconds to open

### ISSUE THREAD

**Issue by filippo1996** [duracelltomi/gtm4wp#349]
Hi, I have a similar problem to this: [](https://wordpress.org/support/topic/product-links-not-working/) 
but in my case the problem concerns the list of products in the categories, when I try to open a product it takes seconds before it processes the page loading, removing this tag <span class="gtm4wp_productdata"> or disabling the
Track e-commerce works normally. You can try it yourself: [](https://www.forap.it/categoria-prodotto/abbigliamento-polizia-locale/)

It is very important that the problem is resolved because the user perceives that the site may not work if it takes this time. the problem occurs on both desktop and mobile.

**Comment by jurijc** [comment]
Hi, I found solution for this problem that I have also noticed. 
You need to copy trigger Event - Ecommerce Events GA4  in gtm and remove from Event name select_item|
Then set this trigger for remarketing tag. 
This is bug because on item click this event is sent and adds delay. Should be fixed @duracelltomi 
![image](https://github.com/duracelltomi/gtm4wp/assets/10640769/e363104d-364a-4d69-a024-c0806b454133)

**Comment by duracelltomi** [comment]
The root cause of this behavior is usually something different in your GTM container or on your website.

When someone clicks on a product in a product list, GTM4WP stops the browser from navigating the user to the product detail page so that it can signal GTM to fire tags related to the select_item event. Now two events can happen:
#1 GTM can fire all related tags quickly and GTM4WP can redirect the user to the product detail page quickly
#2 Something slows down the process, in that case GTM4WP instructs GTM to only wait for 2000ms and just THEN will the user redirected to the product detail page

I am managing some Woo sites where case #2 almost never happens, clicks work quickly with GTM4WP.

So there does not seem to be a single "fix" for that as it could be many factors that causes case #2 to happen.
What I could do is to give more control over the timeout, this way, you could set a lower value by accepting less captured data for the select_item event.

**Comment by gmmedia** [comment]
Thank you so much! This fixed my problem.

> Hi, @filippo1996 I have found solution for this problem that I have also noticed. You need to copy trigger Event - Ecommerce Events GA4 in gtm and remove from Event name select_item| Then set this trigger for remarketing tag. This is bug because on item click this event is sent and adds delay. Should be fixed @duracelltomi ![image](https://private-user-images.githubusercontent.com/10640769/334213779-e363104d-364a-4d69-a024-c0806b454133.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUxMDQ2OTgsIm5iZiI6MTcyNTEwNDM5OCwicGF0aCI6Ii8xMDY0MDc2OS8zMzQyMTM3NzktZTM2MzEwNGQtMzY0YS00ZDY5LWEwMjQtYzA4MDZiNDU0MTMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMxVDExMzk1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYwMjk0ZmFjMjQ3ZWE0NjQ1ZDU0ZDc5NDM3ZjU0ZDQyZmNhYWM2ZWIwZWM1ZjRmMTViNjk2MTc4MGM1ZmJmMDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WiVvgl-qj08BWSZCzwb9-kvFNuElOABuUlq0OE1PIao)

**Comment by filippo1996** [comment]
> La causa principale di questo comportamento è solitamente dovuta a qualcosa di diverso nel contenitore GTM o nel sito web.
> 
> Quando qualcuno clicca su un prodotto in un elenco di prodotti, GTM4WP impedisce al browser di indirizzare l'utente alla pagina dei dettagli del prodotto in modo che possa segnalare a GTM di attivare i tag correlati all'evento select_item. Ora possono verificarsi due eventi: #1 GTM può attivare rapidamente tutti i tag correlati e GTM4WP può reindirizzare rapidamente l'utente alla pagina dei dettagli del prodotto #2 Qualcosa rallenta il processo, in quel caso GTM4WP ordina a GTM di attendere solo 2000 ms e SOLO ALLORA l'utente verrà reindirizzato alla pagina dei dettagli del prodotto
> 
> Gestisco alcuni siti Woo in cui il caso [n. 2](https://github.com/duracelltomi/gtm4wp/issues/2) non si verifica quasi mai, i clic funzionano rapidamente con GTM4WP.
> 
> Quindi non sembra esserci una singola "soluzione" per questo, poiché potrebbero esserci molti fattori che causano il verificarsi del caso [n](https://github.com/duracelltomi/gtm4wp/issues/2) . 2. Ciò che potrei fare è dare più controllo sul timeout, in questo modo, potresti impostare un valore inferiore accettando meno dati catturati per l'evento select_item.

Hi, I'm using Hello Elementor as a theme with the Elementor plugin. I've also tried disabling the other plugins but the problem persists. I don't understand what this could be due to.

**Comment by Axos11** [comment]
Hello @duracelltomi ,
first of all thanks for the amazing plugin!

I encounter the issue, that I have the full 2000ms second delay when the consent is not givven via cmp (I use truendo with consent mode implemented). That immediatlely goes to basically nothing as soon as the consent is granted. 

Any ideas on how to fix that? I basically only have the cmp tag and the google tags (ga4 gads remarketing..) in the container as of now

**Comment by Axos11** [comment]
@duracelltomi 

I mean it is just an implementation of the consent mode and if the consent isn't given it seems that the call won't return. So the question is if it is possible to check for the advertising consent state and not trigger the event if it is disabled or if the 2000ms could be set dynamically via a setting filter to 500ms e.g. Even though I don't like the approach of delaying every click by 500ms unnecessarily. 

Example to test the behaviour: https://www.birkengold.com/pr/xylit-kaugummi-icemint
Basically it happens on every "loop" product listing when advertising is denied.

**Comment by alpharev** [comment]
> Hi,
> 
> Unfortunately I do not have any experience with the mentioned CMP.

Hello
I have the same issue, if user has not accepted cookies consent, it ads delay on product pages. I am using tag manager to implement cookies ( https://support.google.com/tagmanager/answer/14565698) with https://cookieinformation.com/ tag on tag manager.

Please do you have any solution to solve this?

**Comment by Axos11** [comment]
@duracelltomi any hopes of a discussion here? In my oppinion that has to affect more people than just me and @alpharev but I assume as nothing is "completely broken" just strongly delayed (which leads to a horrible experience) they won't recognize it?

I mean a 2000 ms second delay artificially is really sad.

I assume there would be the solution to just let it timeout via a variable that could be e.g. set to 500ms but still would slow down a process that shouldn't be slowed down.

I think the better solution would be to check for the consent mode variables and just don't run or exit the function directly before trying to send on those sites. 

Open for a discussion on that matter?

**Comment by gmmedia** [comment]
Use the workaround from @jurijc 
Otherwise, is GTM4WP not usable.

**Comment by duracelltomi** [comment]
I've added a new plugin option that will enable everyone to customize the default 2000ms timout.

If you can test it, I am happy to receive feedback

**Comment by alpharev** [comment]
> I've added a new plugin option that will enable everyone to customize the default 2000ms timout.
> 
> If you can test it, I am happy to receive feedback
Where is placed that option on the pluggin settings?

**Comment by duracelltomi** [comment]
It is added on the Integration => WooCommerce tab in GTM4WP options.
You can download the beta from here: https://github.com/duracelltomi/gtm4wp/releases/tag/1.22.beta1

**Comment by Axos11** [comment]
@duracelltomi Thanks for the adaption. I just yesterday installed the update on one of my sites and actually had the feeling that nothing did change even though i set the value to 500ms in the backend.

<img width="877" height="600" alt="Image" src="https://github.com/user-attachments/assets/db070ea4-bb40-4354-bee9-24a13229261e" />

Somehow the script seems not to access the correct value as it is set correctly when i check in the console but when i debug the script it is still 2000ms. Maybe someone has an idea why thats happening?

I will try to find the underlaying issue of why the delay happens in the first place.

**Comment by duracelltomi** [comment]
Thanks for the report — and for the useful detail that removing the `.gtm4wp_productdata` span or disabling e-commerce tracking makes it go away; that pinpoints it.

The cause is the `select_item` click handler: so GTM can fire its tags before the browser leaves the list page, it briefly waits for GTM's `eventCallback` (up to the **"Set maximum timeout for select_item event"** value — default 2000&nbsp;ms). When that callback never returns (for example advertising consent is denied, so the tags never fire), the wait runs to the full timeout — which is the delay you're seeing.

This is already configurable, no code change needed on your side:

- Set **WooCommerce → Advanced → "Set maximum timeout for select_item event"** to **`0`** → the click opens the product **immediately**; the `select_item` event is still pushed to the data layer, the navigation just isn't held back.
- Or set it to a value **lower than 2000** (e.g. 500) to shorten the worst-case wait while still giving tags a moment to fire.

The `0` behaviour wasn't spelled out before, so I've clarified that option's description in 2.0 (5b97eab). Closing as resolved via the existing setting — please try setting it to `0` and reopen if you still see a delay.
